core/src/main/java/bio/terra/pearl/core/service/export/formatters/SurveyFormatter.java

```diff
@@ -114,6 +121,17 @@ protected String stripStudyPrefixes(String stableId) {
         return stableId.substring(stableId.lastIndexOf('_') + 1);
     }
 
+    public ItemExportInfo getItemExportInfo(ExportOptions exportOptions, String moduleName, List<SurveyQuestionDefinition> questionDefs)
+            throws JsonProcessingException {
```

Comment:
What is the HTTP response code associated with `JsonProcessingException`?

Reply:
That Exception would mean there's something in a previously saved survey definition that contains an error.  The root line that throws this is line 139: `objectMapper.readValue(questionDef.getChoices(), new TypeReference<List<QuestionChoice>>(){})`  So if somehow some bad choices previously snuck past our on-save validation, that would error here.  I think 500 is the correct error because it's an error not related to the client request for export.  That's also why I thought just catching that error lower down and rethrowing it as a runtime exception might be appropriate.  Essentially "this study has corrupt survey definitions, we don't know what to do"

Reply:
I'm probably hypersensitive to exception handling since we had to spend a lot of effort to get that cleaned up in Pepper, and we are still not done. It really helps to get it right from the get go.

Catching and rethrowing here is provides an opportunity to say exactly what you said above ("this study has corrupt survey definitions..."), unless that is easily deduced as is. 

I'm not a fan of throwing RuntimeException, since developers may throw that in a variety of situations. How about creating two exception classes like `JuniperInternalError` and `BadRequestException`? Those allow developers to unambiguously choose the upstream handling (in most cases that means choosing the HTTP response code). Presumably there is a way for us to map these exceptions to HTTP response handling.

This also provides an opportunity to log and strip information from internal exceptions, since we may not want to share that with end users.

Reply:
+1 for improving exception handling in general. I'm definitely guilty of throwing RuntimeExceptions in places where I should've been more specific. It might be worth creating a tech debt/developer's choice ticket to spend a bit of time auditing our current exception handling and improving it, while the size of the app is still manageable.